validator/project_validator_test.go

```diff
@@ -649,22 +649,74 @@ buildvariants:
 }
 
 func TestCheckRequestersForTaskDependencies(t *testing.T) {
-	t.Run("DependingOnNonPatchableTaskGeneratesWarning", func(t *testing.T) {
-		p := model.Project{
-			Tasks: []model.ProjectTask{
-				{Name: "t1", DependsOn: []model.TaskUnitDependency{
-					{Name: "t2", Variant: model.AllVariants},
-				}},
-				{Name: "t2", Patchable: utility.FalsePtr()},
-			},
-		}
-		allTasks := p.FindAllTasksMap()
-		errs := checkRequestersForTaskDependencies(&p.Tasks[0], allTasks)
-		errs = append(errs, checkRequestersForTaskDependencies(&p.Tasks[1], allTasks)...)
-		require.Len(t, errs, 1)
-		assert.Equal(t, Warning, errs[0].Level)
-		assert.Contains(t, errs[0].Message, "'t1' depends on non-patchable task 't2'")
-	})
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	assert := assert.New(t)
+	exampleYml := `
+exec_timeout_secs: 3600
+tasks:
+  - name: dep1
+    patchable: false
+    commands:
+      - command: shell.exec
+        params:
+          shell: bash
+          script: |
+            echo "hi"
+
+  - name: dep2
+    commands:
+      - command: shell.exec
+        params:
+          shell: bash
+          script: |
+            echo "hi"
+      
+  - name: dep3
+    commands:
+      - command: shell.exec
+        params:
+          shell: bash
+          script: |
+            echo "hi"
+  
+  - name: task1
+    patchable: true
+    depends_on:
+      - name: dep1
+      - name: dep2
+      - name: dep3
+    commands:
+      - command: shell.exec
+        params:
+          shell: bash
+          script: |
+            echo "hi"
+
+buildvariants:
+  - name: ubuntu2204
+    display_name: Ubuntu 22.04
+    patchable: false
+    run_on:
+      - localhost
+    tasks:
+      - name: "task1"
+      - name: "dep1"
+      - name: "dep2"
+        patchable: false
+      - name: "dep3"
```

Comment:
Would it be possible to split this test down into simpler YAML tests by making more test case functions, so that it doesn't get too tricky to understand what's being tested? It seems like this one test case is testing multiple different conditions here, each of which results in a validation warning for different reasons (one because `dep1` sets patchable on the task level, one because `dep2` sets `patchable` is set at the BVTU level, and one because `dep3` inherits `patchable` from the BV level). It's generally preferable to try keeping test cases small and focused on testing a particular condition, rather than trying to cover as many cases as possible within a single test, since splitting up test cases reduces complexity.

Reply:
Sure, done